src/nidigital/unit_tests/test_nidigital.py

```diff
@@ -36,11 +36,68 @@ def teardown_method(self, method):
         self.side_effects_helper['GetHistoryRAMSampleCount']['sampleCount'] = 7
         with nidigital.Session('') as session:
             with pytest.raises(ValueError, match='position: Specified value = 8, Maximum value = 6.'):
                 session.sites[1].fetch_history_ram_cycle_information(position=8, samples_to_read=-1)
+
+    # TODO(sbethur): When nidigital driver provides better simulation support (internal bug# 992370),
+    #  this test should be converted to a system test.  
+    def test_fetch_history_ram_cycle_information_is_finite_invalid(self):
+
+        self.patched_library.niDigital_GetHistoryRAMSampleCount.side_effect = self.side_effects_helper.niDigital_GetHistoryRAMSampleCount
+        self.side_effects_helper['GetHistoryRAMSampleCount']['sampleCount'] = 7
+        self.patched_library.niDigital_GetAttributeViBoolean.side_effect = self.side_effects_helper.niDigital_GetAttributeViBoolean
+        self.side_effects_helper['GetAttributeViBoolean']['value'] = False # history_ram_number_of_samples_is_finite
+
+        expected_error_description = (
+            'Specifying -1 to fetch all History RAM samples is not supported when the digital pattern instrument '
+            'is configured for continuous History RAM acquisition. You must specify an exact number of samples to fetch.')
+        with nidigital.Session('') as session:
+            with pytest.raises(RuntimeError, match=expected_error_description):
+                session.sites[1].fetch_history_ram_cycle_information(position=0, samples_to_read=-1)
+
+    # TODO(sbethur): When nidigital driver provides better simulation support (internal bug# 992370),
+    #  this test should be converted to a system test.  
+    def test_fetch_history_ram_cycle_information_samples_to_read_too_much(self):
+
+        self.patched_library.niDigital_GetHistoryRAMSampleCount.side_effect = self.side_effects_helper.niDigital_GetHistoryRAMSampleCount
+        self.side_effects_helper['GetHistoryRAMSampleCount']['sampleCount'] = 7
+        self.patched_library.niDigital_GetAttributeViBoolean.side_effect = self.side_effects_helper.niDigital_GetAttributeViBoolean
+        self.side_effects_helper['GetAttributeViBoolean']['value'] = True # history_ram_number_of_samples_is_finite
+
+        with nidigital.Session('') as session:
+            assert session.sites[1].get_history_ram_sample_count() == 7
+
+            expected_error_description = (
+                'position: Specified value = 3, samples_to_read: Specified value = 5; Samples available = 4.')
+            with pytest.raises(ValueError, match=expected_error_description):
+                session.sites[1].fetch_history_ram_cycle_information(position=3, samples_to_read=5)
+
+    # TODO(sbethur): When nidigital driver provides better simulation support (internal bug# 992370),
+    #  this test should be converted to a system test.  
+    def test_fetch_history_ram_cycle_information_samples_to_read_zero(self):
```

Comment:
This test can be run on simulated session as system test. You can remove this unit test.

Reply:
Removed